### ISSUE TITLE
Secretary doesn't comply to URL syntax specification

### ISSUE THREAD

**Issue by piotr-yuxuan** [gf3/secretary#91]
Hiya there,

First of all, thank you very much for that piece of free software :-) I've been using happily for a long time and it works well expect for this edge case which has been rather painful for me. To the best of my knowledge there is no issue about this so I fill this one just to reference this behaviour.

According to Wikipedia (and explained in a very circumlocutory way on https://url.spec.whatwg.org/) the syntax of a URL is :
``` uri
scheme:[//[user:password@]host[:port]][/]path[?query][#fragment]
```

However, when secretary is used without html5 routing the in-app route is stored in the fragment and then the query part is added. Reasons for this are obvious (legibility) but resulting URL respect no longer the syntax:

``` uri
scheme:[//[user:password@]host[:port]][/]path[#fragment][?query]
```

Using this custom syntax will cause Secretary and Standard libraries to firmly disagree on some painful edge cases, e.g. : 
``` uri
https://www.example.com/path/of/app#path/inside/app?query=params&as=defined&by=secretary
```

- fragment: `"path/inside/app?query=params&as=defined&by=secretary"` for standard libraries but should be `"path/inside/app"` according to Secretary
- query: `""` but should be `"query=params&as=defined&by=secretary"` according to Secretary

Of course, a very naive regex will get us data we need:

``` clojure
(def hash-according-to-secretary-syntax
  (comp second (partial re-find #"\#([^?]*)"))
```

I just wonder if you could add tiny functions somewhere in a utility namespace to allow Secretary's users to access fragment and query as meant by the custom syntax used. If you agree with this but don't have much time to spare on this detail, just give me general directions about where / how put new code and I'll be happy to do that.

**Comment by gf3** [comment]
this certainly makes sense to me; unfortunately i definitely don't have time to address something like this anytime soon

**Comment by piotr-yuxuan** [comment]
Yeah, we all have either ideas but no time or time but no ideas ^^ I would happen to have not much time and even fewer ideas but I'd be glad to give it a try and see if I can contribute on that.

I've got three possible paths in mind, however I don't have a lot of experience in design so I wouldn't dare say one is _correct_:

- Just add utility functions to get fragment, query parameters a la Secretary
- Add an option (where?) to make secretary favour either the spec or the legibility
``` clojure
;; favour legibility
"https://www.example.com/path/of/app#path/inside/app?query=params&as=defined&by=secretary"

;; favour to comply with spec
"https://www.example.com/path/of/app?query=params&as=defined&by=secretary#path/inside/app"
```
- Write in the readme that Secretary doesn't comply with spec and html5 history should be used with [Accountant](https://github.com/venantius/accountant)
